Receiver/Receiver.c

```diff
@@ -0,0 +1,342 @@
+\param[out]  movingAvg  : pointer to moving average of parameter
+*//*--------------------------------------------------------------------------
+A U T H O R  I D E N T I T Y                                             *//*!
+\author   $Gagandeep
+*//**************************************************************************/
+void simpleMovingAvg(float * previousIndividualParValueSet, float *movingAvg) {
+
+  *movingAvg = 0.f;
+  float sumOfParValue = 0.f;
+  /* Running loop for movingAvgDataCount no. of times for evaluation */
+  for (int CounterOfParValue = 0; CounterOfParValue< movingAvgDataCount; CounterOfParValue++) {
+   
+    sumOfParValue += previousIndividualParValueSet[CounterOfParValue];
+
+  }// end of loop
+
+  /*Update the average of latest movingAvgDataCount values*/
+  *movingAvg = sumOfParValue / movingAvgDataCount;
+}
+
+/*****************************************************************************
+F U N C T I O N    D E S C R I P T I O N
+------------------------------------------------------------------------------
+Function name:  Receivermain                                            *//*!
+\brief       Main recevier functionality which execute the entire flow:
+             -> reading input streaming data from console 
+             -> extracting the parameter from input stream string
+             -> Calculates Min and Max 
+             -> Calculates simple moving average             
+*//*--------------------------------------------------------------------------
+I N P U T                                                               *//*!
+\param[in] -
+\param[in] -
+*//*--------------------------------------------------------------------------
+O U T P U T                                                             *//*!
+\param[out] -
+*//*--------------------------------------------------------------------------
+A U T H O R  I D E N T I T Y                                             *//*!
+\author   $Gagandeep
+*//**************************************************************************/
+void Receivermain()
+{
+  char paramSet[lengthOfParamset];
+  float previousParamValueBuffer[NoOfPar][movingAvgDataCount] = { 0.f };
+  float paramMin[NoOfPar] = { TempMax,socMax };
+  float paramMax[NoOfPar] = { TempMin,socMin };
+  float movingAvg[NoOfPar] = {0.f};
+
+  int readFailureCounter = 0;
+  
+  //Initialise the global variable
+  communicationfailure = false;
+
+  
+  /* Reading the input stream */
+  for (int paramSetCounter = 0; paramSetCounter < NoOfParamterSet; paramSetCounter++) {
+    
+    /* Read one input set if the read is successful process parameters further*/
+    bool ifTrueReadSuccess = readInputFromConsole(paramSet);
+    if (ifTrueReadSuccess)
+    {
+      /* Initialize individual parameter buffer to zero */
+      float paramBuffer[NoOfPar] = { 0.f };
+
+      /*Converts data stream to individual parameter data*/
+      extractParametersFromInputSet(paramSet, paramBuffer);
```

Comment:
This is a complex function and hard to test. Maybe that's why the test for this function just checks that it didn't crash.
In particular, if we need to test the situation in which `extractParametersFromInputSet` cannot do an atof (due to an error in the input) - it will be hard. Result of complexity is difficulty in testing.